### PR TITLE
Add constants (#41)

### PR DESCRIPTION
**Pull request by MobiliteDev**
I've added all the constants, renamed the class name with "Sql" prefix.
Added to the sqlite3.dart for public visibility

### CODE REVIEW

**Comment by MobiliteDev** [comment]
@simolus3 these CI errors are on your side I think.

image

File: sqlite3/lib/src/ffi/constants.dart [hunk 1]
```diff
@@ -1,44 +1,247 @@
...
+  static const int SQLITE_DONE = 101;
+}
+
+/// Flags for file open operations, https://www.sqlite.org/c3ref/c_open_autoproxy.html
+/// Flags For File Open Operations
+///
+/// These bit values are intended for use in the
+/// 3rd parameter to the [sqlite3_open_v2()] interface and
+/// in the 4th parameter to the [sqlite3_vfs.xOpen] method.
+class SqlFlags {
```
Comment (simolus3) [inline sqlite3/lib/src/ffi/constants.dart:114]: Do we need to add those flags to the public api? I think we don't provide an api where users could pass one of them, so I think we shouldn't make them public.

File: sqlite3/lib/src/ffi/constants.dart [hunk 2]
```diff
@@ -1,44 +1,247 @@
...
+  static const int SQLITE_OPEN_PRIVATECACHE = 0x00040000;
+
+  /// VFS only
+  static const int SQLITE_OPEN_WAL = 0x00080000;
+}
 
 // Prepare flags, https://www.sqlite.org/c3ref/c_prepare_normalize.html
-const SQLITE_PREPARE_PERSISTENT = 0x01;
-const SQLITE_PREPARE_NO_VTAB = 0x04;
+class SqlPrepareFlags {
```
Comment (simolus3) [inline sqlite3/lib/src/ffi/constants.dart:177]: Same here I think?

File: sqlite3/lib/src/ffi/constants.dart [hunk 3]
```diff
@@ -1,44 +1,247 @@
...
+  ///The SQLITE_PREPARE_NORMALIZE flag is a no-op. This flag used to be required for any prepared statement that wanted to use the sqlite3_normalized_sql() interface.
+  /// However, the sqlite3_normalized_sql() interface is now available to all prepared statements, regardless of whether or not they use this flag.
+  static const int SQLITE_PREPARE_NORMALIZE = 0x02;
+
+  ///the SQL compiler to return an error (error code SQLITE_ERROR) if the statement uses any virtual tables
+  static const int SQLITE_PREPARE_NO_VTAB = 0x04;
+}
+
+/// Datatypes, https://sqlite.org/c3ref/c_blob.html
+class SqlTypes {
```
Comment (simolus3) [inline sqlite3/lib/src/ffi/constants.dart:193]: This one is exposed through Dart types, users shouldn't see those.

File: sqlite3/lib/src/ffi/constants.dart [hunk 4]
```diff
@@ -1,44 +1,247 @@
...
+  static const int SQLITE_UTF16_ALIGNED = 8;
+}
 
-const SQLITE_INTEGER = 1;
-const SQLITE_FLOAT = 2;
-const SQLITE_TEXT = 3;
-const SQLITE_BLOB = 4;
-const SQLITE_NULL = 5;
+/// Special destructors, https://www.sqlite.org/c3ref/c_static.html
+class SqlSpecialDestructor {
```
Comment (simolus3) [inline sqlite3/lib/src/ffi/constants.dart:224]: Nice call to introduce fields for these values! Let's just not expose them to users.

File: sqlite3/lib/src/ffi/constants.dart [hunk 5]
```diff
@@ -1,44 +1,247 @@
...
+/// Result Codes
+///
+/// Many SQLite functions return an integer result code from the set shown
+/// here in order to indicates success or failure.
+///
+/// New error codes may be added in future versions of SQLite.
+///
+/// See also: SQLITE_IOERR_READ | extended result codes,
+/// sqlite3_vtab_on_conflict() SQLITE_ROLLBACK | result codes.
+class SqlErrors {
```
Comment (simolus3) [inline sqlite3/lib/src/ffi/constants.dart:13]: Let's call this class `SqlError` which makes it easier to read a specific constant in code.

**Review by simolus3 — COMMENTED** [review]
My server should be fixed now, sorry for that.

I think we should not expose all of these constants since most of them aren't actually useful for users. You can add a `show SqlError` to only expose those.

File: sqlite3/lib/src/ffi/constants.dart (hunk 1 above)
Comment (MobiliteDev) [reply sqlite3/lib/src/ffi/constants.dart]: You are right, I've changed the visibility in sqlite3.dart on the last commit

File: sqlite3/lib/src/ffi/constants.dart (hunk 2 above)
Comment (MobiliteDev) [reply sqlite3/lib/src/ffi/constants.dart]: You are right, I've changed the visibility in sqlite3.dart on the last commit

File: sqlite3/lib/src/ffi/constants.dart (hunk 3 above)
Comment (MobiliteDev) [reply sqlite3/lib/src/ffi/constants.dart]: You are right, I've changed the visibility in sqlite3.dart on the last commit

File: sqlite3/lib/src/ffi/constants.dart (hunk 4 above)
Comment (MobiliteDev) [reply sqlite3/lib/src/ffi/constants.dart:487]: You are right, I've changed the visibility in sqlite3.dart on the last commit

File: sqlite3/lib/src/ffi/constants.dart (hunk 5 above)
Comment (MobiliteDev) [reply sqlite3/lib/src/ffi/constants.dart]: You are right, I've renamed to SqlError

**Comment by MobiliteDev** [comment]
I've removed the 's' from all my classes.

and I've added SqlExtError in public visibility, you get it with `sqlite3_extended_result_codes` function in `database.dart`

**Comment by simolus3** [comment]
I think the CI failures are due to the Flutter dev SDK not having the necessary Dart version yet. Thanks for your contribution, this looks good to me :+1:

**Pull request merged**